web-api/src/getPersistenceGateway.js

```diff
@@ -0,0 +1,651 @@
+} = require('../../shared/src/persistence/dynamo/users/updateUserEmail');
+const {
+  updateWorkItemAssociatedJudge,
+} = require('../../shared/src/persistence/dynamo/workitems/updateWorkItemAssociatedJudge');
+const {
+  updateWorkItemCaseStatus,
+} = require('../../shared/src/persistence/dynamo/workitems/updateWorkItemCaseStatus');
+const {
+  updateWorkItemCaseTitle,
+} = require('../../shared/src/persistence/dynamo/workitems/updateWorkItemCaseTitle');
+const {
+  updateWorkItemDocketNumberSuffix,
+} = require('../../shared/src/persistence/dynamo/workitems/updateWorkItemDocketNumberSuffix');
+const {
+  updateWorkItemTrialDate,
+} = require('../../shared/src/persistence/dynamo/workitems/updateWorkItemTrialDate');
+const {
+  verifyCaseForUser,
+} = require('../../shared/src/persistence/dynamo/cases/verifyCaseForUser');
+const {
+  verifyPendingCaseForUser,
+} = require('../../shared/src/persistence/dynamo/cases/verifyPendingCaseForUser');
+const {
+  zipDocuments,
+} = require('../../shared/src/persistence/s3/zipDocuments');
+const { getDocument } = require('../../shared/src/persistence/s3/getDocument');
+const { getMessages } = require('../../shared/src/persistence/sqs/getMessages');
+
+const isValidatedDecorator = persistenceGatewayMethods => {
+  /**
+   * Decorates the function to verify any entities passed have the isValid flag.
+   * Should be used whenever a persistence method might be called by an interactor via lambda
+   * when an entity's complete record is being created or updated.
+   *
+   * @returns {Function} the original methods decorated
+   */
+  function decorate(method) {
+    return function () {
+      const argumentsAsArray = Array.prototype.slice.call(arguments);
+
+      argumentsAsArray.forEach(argument => {
+        Object.keys(argument).forEach(key => {
+          if (
+            argument[key] &&
+            argument[key].entityName &&
+            !argument[key].isValidated
+          ) {
+            console.trace();
+            throw new Error(
+              `a entity of type ${argument[key].entityName} was not validated before passed to a persistence gateway method`,
+            );
+          }
+        });
+      });
+      return method.apply(null, argumentsAsArray);
+    };
+  }
+
+  Object.keys(persistenceGatewayMethods).forEach(key => {
+    persistenceGatewayMethods[key] = decorate(persistenceGatewayMethods[key]);
+  });
+  return persistenceGatewayMethods;
+};
+
+const gatewayMethods = {
+  ...isValidatedDecorator({
+    addCaseToHearing,
+    associateUserWithCase,
+    associateUserWithCasePending,
+    bulkDeleteRecords,
+    bulkIndexRecords,
+    createCase,
+    createCaseDeadline,
+    createCaseTrialSortMappingRecords,
+    createJobStatus,
+    createMessage,
+    createNewPetitionerUser,
+    createNewPractitionerUser,
+    createOrUpdatePractitionerUser,
+    createOrUpdateUser,
+    createTrialSession,
+    createTrialSessionWorkingCopy,
+    deleteKeyCount,
+    fetchPendingItems,
```

Comment:
this one is also in `getuseCaseHelpers` - is that necessary?

Reply:
by this one i mean `fetchPendingItems`

Reply:
good catch, I think I had some extra persistence files in the wrong file